src/config/theme.ts

```diff
@@ -277,7 +313,63 @@ const createThemeOptions = (theme: Theme) => ({
           // override default transparent bg
           backgroundColor: '#fff',
         },
+        root: ({
+          ownerState,
+          theme,
+        }: {
+          ownerState: AlertProps;
+          theme: Theme;
+        }) => {
+          let bgColor: string = '';
+          let headerColor: string = '';
+          if (ownerState.variant === 'withHeader') {
+            if (!ownerState.color) {
+              throw new Error(
+                'withHeader Alert variant requires color to be specified: low, medium, or high'
+              );
+            } else {
+              bgColor =
+                theme.palette.alerts[
+                  ownerState.color as keyof AlertPropsColorOverrides
+                ].background;
+              headerColor =
+                theme.palette.alerts[
+                  ownerState.color as keyof AlertPropsColorOverrides
+                ].header;
+            }
+          }
+
+          return {
+            ...(ownerState.variant === 'withHeader' && {
+              padding: 0,
+              color: 'black',
+              backgroundColor: bgColor,
+              borderColor: headerColor,
+              '& .MuiAlertTitle-root': {
+                padding: theme.spacing(2),
+                margin: 0,
+                backgroundColor: headerColor,
+                fontWeight: 'bold',
+                color: theme.palette.getContrastText(headerColor),
+              },
+              '& .MuiAlert-message': {
+                padding: 0,
+                width: '100%',
+              },
+              '& .MuiAlert-body': {
+                padding: theme.spacing(2),
+              },
+            }),
+          };
+        },
       },
+      variants: [
```

Comment:
I was playing around with this a bit to see if we can define all these styles in the `variants` section rather than the `root` -- since they are all styles that are specific to one variant. I got it working like this. The `props` key specifies which props should be present for the styles to be applied, and the `style` specifies the styles. There is a way to return sx using `unstable_sx` which I used here.

What do you think? Although there is some repetition, it feels cleaner to me because it's not part of the `root` object -- so we can be sure that it doesn't affect the default alerts.

```
      variants: [
        {
          // styles that are shared across all withHeader alerts
          props: { variant: 'withHeader' },
          style: theme.unstable_sx({
            padding: 0,
            color: 'black',
            '& .MuiAlertTitle-root': {
              padding: theme.spacing(2),
              margin: 0,
              fontWeight: 800,
            },
            '& .MuiAlert-message': {
              padding: 0,
              width: '100%',
            },
            '& .MuiAlert-body': {
              padding: theme.spacing(2),
            },
          }),
        },
        {
          // styles for LOW priority
          props: { variant: 'withHeader', color: 'low' },
          style: theme.unstable_sx({
            borderColor: theme.palette.alerts.low.header,
            backgroundColor: theme.palette.alerts.low.background,
            '& .MuiAlertTitle-root': {
              backgroundColor: theme.palette.alerts.low.header,
              color: theme.palette.getContrastText(
                theme.palette.alerts.low.header
              ),
            },
          }),
        },
        {
          // styles for MEDIUM priority
          props: { variant: 'withHeader', color: 'medium' },
          style: theme.unstable_sx({
            borderColor: theme.palette.alerts.medium.header,
            backgroundColor: theme.palette.alerts.medium.background,
            '& .MuiAlertTitle-root': {
              backgroundColor: theme.palette.alerts.medium.header,
              color: theme.palette.getContrastText(
                theme.palette.alerts.medium.header
              ),
            },
          }),
        },
      ],
```

Reply:
By the way, the `theme` variable referenced here is defined on line `121`. It is the default MUi theme overlayed with any customer-specific values from the backend. The end of the file shows how they are composed

Reply:
Thanks @gigxz this is great, I agree it's a lot cleaner!